library/src/main/java/com/bumptech/glide/RequestBuilder.java

```diff
@@ -903,125 +931,156 @@ private Request buildRequestRecursive(
-          && !thumbnailBuilder.requestOptions.isValidOverride()) {
-        thumbOverrideWidth = requestOptions.getOverrideWidth();
-        thumbOverrideHeight = requestOptions.getOverrideHeight();
-      }
-
-      ThumbnailRequestCoordinator coordinator = new ThumbnailRequestCoordinator(parentCoordinator);
-      Request fullRequest =
-          obtainRequest(
-              target,
-              targetListener,
-              requestOptions,
-              coordinator,
-              transitionOptions,
-              priority,
-              overrideWidth,
-              overrideHeight);
-      isThumbnailBuilt = true;
-      // Recursively generate thumbnail requests.
-      Request thumbRequest =
-          thumbnailBuilder.buildRequestRecursive(
-              target,
-              targetListener,
-              coordinator,
-              thumbTransitionOptions,
-              thumbPriority,
-              thumbOverrideWidth,
-              thumbOverrideHeight,
-              thumbnailBuilder.requestOptions);
-      isThumbnailBuilt = false;
-      coordinator.setRequests(fullRequest, thumbRequest);
-      return coordinator;
+      return buildThumbnailRequestRecursive(thumbnailBuilder,
+          target, targetListener, parentCoordinator, requestOptions,
+          transitionOptions, priority, overrideWidth, overrideHeight);
     } else if (thumbSizeMultiplier != null) {
       // Base case: thumbnail multiplier generates a thumbnail request, but cannot recurse.
-      ThumbnailRequestCoordinator coordinator = new ThumbnailRequestCoordinator(parentCoordinator);
-      Request fullRequest =
-          obtainRequest(
-              target,
-              targetListener,
-              requestOptions,
-              coordinator,
-              transitionOptions,
-              priority,
-              overrideWidth,
-              overrideHeight);
-      RequestOptions thumbnailOptions = requestOptions.clone()
-          .sizeMultiplier(thumbSizeMultiplier);
-
-      Request thumbnailRequest =
-          obtainRequest(
-              target,
-              targetListener,
-              thumbnailOptions,
-              coordinator,
-              transitionOptions,
-              getThumbnailPriority(priority),
-              overrideWidth,
-              overrideHeight);
-
-      coordinator.setRequests(fullRequest, thumbnailRequest);
-      return coordinator;
+      return buildThumbnailRequestForMultiplier(thumbSizeMultiplier,
+          target, targetListener, parentCoordinator, requestOptions,
+          transitionOptions, priority, overrideWidth, overrideHeight);
     } else {
       // Base case: no thumbnail.
       return obtainRequest(
-          target,
-          targetListener,
-          requestOptions,
-          parentCoordinator,
-          transitionOptions,
-          priority,
-          overrideWidth,
-          overrideHeight);
+          target, targetListener, parentCoordinator, requestOptions,
```

Comment:
Style nit: leave the parameters on individual lines.

We use:
method(arg1, arg2, arg3); if the method fits on one line.
method(
    arg1, arg2, arg3); if the method doesn't fit on one line, but the arguments do
method(
    arg1,
    arg2,
    arg3); if the arguments don't fit on one line.

It's fine to leave places where we don't follow this alone, but we shouldn't revert places where we've already switched to doing this.